mapreduce/datadog_checks/mapreduce/mapreduce.py

```diff
@@ -194,6 +196,15 @@ def _parse_job_specific_counters(self, init_config):
 
         return job_counter
 
+    def _get_hadoop_version(self):
+        if self.is_metadata_collection_enabled():
+            try:
+                cluster_info = self._rest_request_to_json(self.rm_address, self.CLUSTER_INFO)
+                hadoop_version = cluster_info.get('clusterInfo', {}).get('hadoopVersion', '')
+                self.set_metadata('version', hadoop_version)
+            except Exception as e:
+                self.log.warning("There was an error retrieving hadoop version {}", e)
```

Comment:
```suggestion
        if not self.is_metadata_collection_enabled():
            return
        try:
            cluster_info = self._rest_request_to_json(self.rm_address, self.CLUSTER_INFO)
            hadoop_version = cluster_info.get('clusterInfo', {}).get('hadoopVersion', '')
            self.set_metadata('version', hadoop_version)
        except Exception as e:
            self.log.warning("There was an error retrieving hadoop version {}", e)
```

(nit) Avoid too many indentation.